cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/BasinEndpointInput.java

```diff
@@ -0,0 +1,148 @@
+package mil.army.usace.hec.cwms.radar.client.controllers;
+
+import static mil.army.usace.hec.cwms.radar.client.controllers.RadarEndpointConstants.ACCEPT_HEADER_V1;
+import static mil.army.usace.hec.cwms.radar.client.controllers.RadarEndpointConstants.ACCEPT_QUERY_HEADER;
+
+import java.util.Objects;
+import mil.army.usace.hec.cwms.http.client.EndpointInput;
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilder;
+import mil.army.usace.hec.cwms.radar.client.model.Basin;
+import mil.army.usace.hec.cwms.radar.client.model.DeleteMethod;
+
+
+public final class BasinEndpointInput {
+    private BasinEndpointInput() {
+        throw new AssertionError("factory class");
+    }
+
+    public static GetAll getAll(String projectId, String officeId) {
+        return new GetAll(projectId, officeId);
+    }
+
+    public static GetOne getOne(String basinId, String officeId) {
+        return new GetOne(basinId, officeId);
+    }
+
+    public static Post post(Basin basin) {
+        return new Post(basin);
+    }
+
+    public static Delete delete(String basinId, String officeId) {
+        return new Delete(basinId, officeId);
+    }
+
+    public static Patch patch(String oldBasinId, String newBasinId, String officeId) {
+        return new Patch(oldBasinId, newBasinId, officeId);
+    }
+
+    public static final class GetAll extends EndpointInput {
+        static final String OFFICE_QUERY_PARAMETER = "office";
+        static final String PROJECT_ID_QUERY_PARAMETER = "project-id";
+        private final String projectId;
+        private final String officeId;
+
+        private GetAll(String projectId, String officeId) {
+            this.projectId = Objects.requireNonNull(projectId, "Project Id required for getAll basin endpoint");
+            this.officeId = Objects.requireNonNull(officeId, "Project office Id required for getAll basin endpoint");
+        }
+
+        @Override
+        public HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(PROJECT_ID_QUERY_PARAMETER, projectId)
+                    .addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class GetOne extends EndpointInput {
+        public static final String OFFICE_QUERY_PARAMETER = "office";
+        private final String officeId;
+        private final String basinId;
+
+        public GetOne(String basinId, String officeId) {
+            this.basinId = Objects.requireNonNull(basinId, "Basin Id required for getOne basin endpoint");
+            this.officeId = Objects.requireNonNull(officeId, "Basin office Id required for getOne basin endpoint");
+        }
+
+        public String basinId() {
```

Comment:
reduce scope to package-private if possible

Reply:
Fixed in latest commit